lib/MessageDecoder.ts

```diff
@@ -106,29 +103,32 @@ export class MessageDecoder {
       console.log(usablePlugins);
     }
 
-    let result : DecodeResult;
-    if (usablePlugins.length > 0) {
-      const plugin: DecoderPluginInterface = usablePlugins[0];
+    let result: DecodeResult = {
+      decoded: false,
+      error: 'No known decoder plugin for this message',
+      decoder: {
+        name: 'none',
+        type: 'none',
+        decodeLevel: 'none',
+      },
+      message: message,
+      remaining: {
+        text: message.text,
+      },
+      raw: {},
+      formatted: {
+        description: 'Not Decoded',
+        items: [],
+      },
+    };
+
+    // for-in is not happy. doing it the old way
+    for (let i = 0; i < usablePlugins.length; i++) {
+      const plugin = usablePlugins[i];
       result = plugin.decode(message);
-    } else {
-      result = {
-        decoded: false,
-        error: 'No known decoder plugin for this message',
-        decoder: {
-          name: 'none',
-          type: 'none',
-          decodeLevel: 'none',
-        },
-        message: message,
-        remaining: {
-          text: message.text,
-        },
-        raw: {},
-        formatted: {
-          description: 'Not Decoded',
-          items: [],
-        },
-      };
+      if (result.decoded) {
+        break;
+      }
```

Comment:
this is important - moving from grabbing the first match, to running in order and stopping if we successfully decoded